tests/test_blockchain_gateway.py

```diff
@@ -49,10 +48,11 @@ def test_blockchain_gateway_can_listen_decentralized_learning(config_manager, co
     blockchain_gateway.configure(config_manager, communication_manager)
     tx_receipt = setter(blockchain_gateway.client, None, blockchain_gateway.port, {"model": "hello world"}, True)
     assert tx_receipt
-    blockchain_gateway.listen_decentralized_learning()
+    blockchain_gateway._listen(blockchain_gateway._handle_new_session_creation,
+        blockchain_gateway._filter_new_session)
     # at this point we should listen for decentralized learning, hear it, and update our communication manager
     assert communication_manager.dummy1 == RawEventTypes.NEW_SESSION.name, "Wrong dummy1"
-    assert communication_manager.dummy2, "No dummy2"
+    assert communication_manager.dummy2 == {"model": "hello world"}, "Wrong dummy2"
```

Comment:
We're using magic strings in both tests, but different magic strings. I want the test to be very easy to follow if a dev fails the test; "what was my output supposed to be?" "ok, that's what it was supposed to be." rather than having to check the datacache to see what we wanted the dummy variable to be equal to.